### ISSUE TITLE
Remove old fields from analysis_options.yaml

### ISSUE THREAD

**Issue by srawlins** [dart-lang/sdk#50680]
I see support in analyzer's `AnalysisOptions` class, support for:

* `enableSuperMixins`
* `enablePreviewDart2`
* `optional-checks` (maybe just the Chrome OS Manifest checks?)
* `strong-mode` (`declaration-casts`, `implicit-casts`, `implicit-dynamic`), covered by https://github.com/dart-lang/sdk/issues/50679
* `propagate-linter-exceptions`

We could remove at least `enableSuperMixins`, `enablePreviewDart2`, and `strong-mode`. Not sure about `optional-checks` or `propagate-linter-exceptions`.

**Comment by srawlins** [comment]
We have already deprecated `enableSuperMixins` (with `AnalysisOptionsHintCode.SUPER_MIXINS_SETTING_DEPRECATED`) and `enablePreviewDart2` (with `AnalysisOptionsHintCode.PREVIEW_DART_2_SETTING_DEPRECATED`).

**Comment by srawlins** [comment]
cc @pq do you know what `propagate-linter-exceptions` is?

**Comment by pq** [comment]
Yeah.  This is something @stereotype441 added to improve internal builds and testing (more context: https://dart-review.googlesource.com/c/sdk/+/208141).

**Comment by srawlins** [comment]
Ok cool, we'll keep it!
